### PR TITLE
[FLINK-34118] Implement restore tests for Sort node

### PR DESCRIPTION
**Pull request by bvarghese1**
## What is the purpose of the change

*Add restore tests for Sort node*

## Verifying this change

This change added tests and can be verified as follows:

- Added restore tests for Sort node which verifies the generated compiled plan with the saved compiled plan.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/SortTestPrograms.java [hunk 1]
```diff
@@ -123,4 +125,66 @@ public class SortTestPrograms {
...
                     .runSql("INSERT INTO sink_t SELECT * from source_t ORDER BY a DESC LIMIT 3")
                     .build();
+
+    static final TableTestProgram SORT_ASC =
+            TableTestProgram.of("sort-asc", "validates sort node by sorting integers in asc mode")
+                    .setupConfig(InternalConfigOptions.TABLE_EXEC_NON_TEMPORAL_SORT_ENABLED, true)
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema("a INT", "b VARCHAR", "c INT")
+                                    .addOption("terminating", "true")
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/SortTestPrograms.java:135]: Is this really necessary? If so, can you please put an explanation why/

File: flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/SortTestPrograms.java [hunk 2]
```diff
@@ -123,4 +125,66 @@ public class SortTestPrograms {
...
+                    .runSql("INSERT INTO sink_t SELECT * from source_t ORDER BY a")
+                    .build();
+
+    static final TableTestProgram SORT_DESC =
+            TableTestProgram.of("sort-desc", "validates sort node by sorting integers in desc mode")
+                    .setupConfig(InternalConfigOptions.TABLE_EXEC_NON_TEMPORAL_SORT_ENABLED, true)
+                    .setupTableSource(
+                            SourceTestStep.newBuilder("source_t")
+                                    .addSchema("a INT", "b VARCHAR", "c INT")
+                                    .addOption("terminating", "true")
```
Comment (dawidwys) [inline flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/SortTestPrograms.java:166]: same as above

Comment (bvarghese1) [reply flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/SortTestPrograms.java]: Removed by converting to a non restore test

**Review by dawidwys — APPROVED** [review]
lgtm

**Pull request merged**